src/Entity/Game.php

```diff
@@ -45,4 +45,20 @@ public function addShip(Ship $ship): self
 
         return $this;
     }
+
+    public function getShipAt(int $x, int $y)
```

Comment:
Please move this logic outside the entity. Entity need to be more "stupid" as possible